runtime/amd64.S

```diff
@@ -635,13 +630,19 @@ CFI_STARTPROC
     /* Make the alloc ptr available to the C code */
         movq    %r15, Caml_state(young_ptr)
     /* Copy arguments from OCaml to C stack */
+#if defined(SYS_mingw64) || defined (SYS_cygwin)
+        addq    $32, %rsp
+#endif
```

Comment:
I understand what you're doing here, but maybe a comment is warranted, something like
"Temporarily discard the reserved 32 bytes at bottom of stack while copying the arguments".